src/main/java/uk/gov/companieshouse/company/links/service/DeleteStatementsClient.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.companieshouse.company.links.service;
+
+import java.util.function.Supplier;
+
+import org.springframework.stereotype.Component;
+
+import uk.gov.companieshouse.api.InternalApiClient;
+import uk.gov.companieshouse.api.error.ApiErrorResponseException;
+import uk.gov.companieshouse.api.handler.exception.URIValidationException;
+import uk.gov.companieshouse.company.links.exception.NonRetryableErrorException;
+import uk.gov.companieshouse.company.links.exception.RetryableErrorException;
+import uk.gov.companieshouse.logging.Logger;
+
+@Component
+public class DeleteStatementsClient implements LinkClient {
+    private final Logger logger;
+    private final Supplier<InternalApiClient> internalApiClientFactory;
+
+    public DeleteStatementsClient(Logger logger, Supplier<InternalApiClient> internalApiClientFactory) {
+        this.logger = logger;
+        this.internalApiClientFactory = internalApiClientFactory;
+    }
+
+    @Override
+    public void patchLink(String companyNumber) {
+        InternalApiClient client = internalApiClientFactory.get();
+        try {
+            client.privateCompanyLinksResourceHandler().deletePscStatementsCompanyLink(String.format("/company/%s/links/persons-with-significant-control-statements/delete", companyNumber)).execute();
```

Comment:
Could we first get the number of psc statements from mongo and confirm that they are all deleted before deleting the link?

Use the getStatements method from the sdk to make an api call